### ISSUE TITLE
bug: presence of a TaskWork doesn't prevent the associated TaskPlan from showing in active tasks

### ISSUE THREAD

**Issue by iainkirkpatrick** [root-systems/cobuy#228]
PR #276

**Comment by sarah-arrrgh** [comment]
**TaskRecipe** = What is this task about?
(lives in tasks/data/recipes.js, eventually will live in database)

If we want a user to create their profile, we give them a taskPlan, with the correct taskRecipe id

e.g.
```
export const createProfile = {
  id: 'createProfile',
  Component: CreateProfileTask,
  childTaskRecipes: []
}
```

---

**TaskPlan** = Who is this task assigned to?
(lives in database)
A record that links a user to a taskRecipe

---

**TaskWork** = A record that the associated task plan is done
(lives in database)

User has a taskPlan called createProfile. When they have created their profile, we create a taskWork record that links to that taskPlan id.

**Comment by sarah-arrrgh** [comment]
- [x] for a previously completed task - are we getting the TaskWork in state on the dashboard? - YES
- [ ] let's not show the TaskPlans that have a TaskWork associated with them 🗡

**Comment by sarah-arrrgh** [comment]
## Sarah's brain thoughts:

- [ ] wiring up a getter somewhere to feed only TaskPlans that **do not** have a TaskWork id associated with them

Tree of my mind:

- (component) DashboardTasks - `renderParentTasks` from `taskPlans`
- (component) Dashboard
- (container) Dashboard
- (getter) getDashboardProps
- (getter) getParentTaskPlans
- (getter) getTaskPlans


Currently all TaskPlans are "gotten" and fed through. Currently we only need TaskPlans that need completing to be fed through... 

Currently two getters - `getParentTaskPlans` and `getActiveTaskPlans` - seems like the latter is what we need to rewrite and feed through...

**Comment by sarah-arrrgh** [comment]
```
const getParentTaskPlans = createSelector(
  getTaskPlans,
  pipe(
    tap( (x) => {
      console.log('pipe stuff (what getTaskPlans is returning): ', x)
    }),
    values,
    tap( (x) => {
      console.log('pipe values: ', x)
    }),
    filter(pipe(
      tap( (x) => {
      console.log('middle of pipe stuff (start of filter): ', x)
    }),
      prop('parentTaskPlanId'),
      isNil
    )),
    tap( (x) => {
      console.log('end of pipe stuff (what getParentTaskPlans is returning): ', x)
    })
  )
)
```

**Comment by sarah-arrrgh** [comment]
### ramda filter

two arguments: 

1 - predicate - a rule that returns true or false

2 - array - that the rule will be applied to

returns an array or subarray of all the array bits that conform to the rule

**Comment by sarah-arrrgh** [comment]
Mikey says:

- do ramda things the ramda way
- don't give a thing two arguments at once, give it the first one first, then that returns a function that you can call with the data later on.
- instead of having a separate getter to figure out which taskPlans are active, do the same logic in the enchancedTaskPlans getter that adds a thing to the taskPlans state called `isActive` that can be true or false depending the result of that logic...

**Comment by sarah-arrrgh** [comment]
@iainkirkpatrick and @sarah-arrrgh were pairing on this on thursday morning. hit bugs when burnthedb. need to `git stash pop` on sarah's machine

**Comment by michael-smith-nz** [comment]
Hey @iainkirkpatrick  @sarah-arrrgh I just tested this PR and it works!.... but adds another problem.

I pulled master, burned the db and now parent AND child tasks are showing on the dashboard which was very confusing as I was expecting to see only tasks without a parent task.
![screenshot from 2017-11-09 11-33-18](https://user-images.githubusercontent.com/1152577/32603566-15dfc2ca-c542-11e7-8f5a-2db5fe71c3ae.png)

I attempted to fix this on #282
